### PR TITLE
API (create_parameter, create_global_var) error message enhancement

### PR DESCRIPTION
**Pull request by FrostML**
API (create_parameter) error message enhancement. 
API (create_global_var) error message enhancement. 
Since create_parameter and create_global_var are both used for creating tensor, then put them into one PR.

### CODE REVIEW

File: python/paddle/fluid/layers/tensor.py [hunk 1]
```diff
@@ -136,6 +156,22 @@ def create_global_var(shape,
             var = layers.create_global_var(shape=[2,3], value=1.0, dtype='float32',
                                           persistable=True, force_cpu=True, name='new_var')
     """
+    check_type(shape, 'shape', (list, tuple), 'create_global_var')
+    for item in shape:
+        if six.PY2:
+            check_type(item, 'item of shape',
+                       (int, long, numpy.uint8, numpy.int8, numpy.int16,
+                        numpy.int32, numpy.int64), 'create_parameter')
```
Comment (guoshengCS) [inline python/paddle/fluid/layers/tensor.py]: Should `create_parameter ` be `create_global_variable`

File: python/paddle/fluid/layers/tensor.py [hunk 2]
```diff
@@ -136,6 +156,22 @@ def create_global_var(shape,
...
+        else:
+            check_type(item, 'item of shape',
+                       (int, numpy.uint8, numpy.int8, numpy.int16, numpy.int32,
+                        numpy.int64), 'create_parameter')
```
Comment (guoshengCS) [inline python/paddle/fluid/layers/tensor.py]: should `create_parameter ` be `create_global_variable`

File: python/paddle/fluid/layers/tensor.py (hunk 1 above)
Comment (FrostML) [reply python/paddle/fluid/layers/tensor.py]: Oh! Done. Thanks.

File: python/paddle/fluid/layers/tensor.py (hunk 2 above)
Comment (FrostML) [reply python/paddle/fluid/layers/tensor.py]: Done. Thanks.

File: python/paddle/fluid/layers/tensor.py [hunk 3]
```diff
@@ -97,10 +99,28 @@ def create_parameter(shape,
...
+
+    check_dtype(dtype, 'dtype', ['float16', 'float32', 'float64'],
```
Comment (guoshengCS) [inline python/paddle/fluid/layers/tensor.py]: Is it a good choice to restrict parameters to be 'float16', 'float32', 'float64'

Comment (FrostML) [reply python/paddle/fluid/layers/tensor.py]: Done. Thanks. If the `dtype` is `int`, then it's necessary to set `default_initializer` and this works fine.

**Review by guoshengCS — APPROVED** [review]

**Pull request merged**